package.json

```diff
@@ -89,7 +89,13 @@
       "index.ts",
       ".+\\/__tests__\\/.+\\.factory.(t|j)s"
     ],
-    "testEnvironment": "node"
+    "testEnvironment": "node",
+    "moduleNameMapper": {
+      "^@domain/(.*)$": "<rootDir>/../src/domain/$1",
+      "^@data/(.*)$": "<rootDir>/../src/datasources/$1",
+      "^@routes/(.*)$": "<rootDir>/../src/routes/$1",
```

Comment:
I would sooner suggest `@/` as an alias. There is otherwise overlap with packages, e.g. `@safe-global/safe-deployments`.

Reply:
Interesting, I haven't thought about that. Afaik the `@` notation refers to scoped packages but I am not sure if there are any restrictions on them i.e. if someone could publish a package with scope `@domain` or `@routes`. Can we somehow test how this could be an issue with an example?

Reply:
You could install a package that begins with one of the specified aliases.

I don't think there should be a differing prefix though. It's [currently possible](https://github.com/safe-global/safe-client-gateway-nest/blob/path-alias/tsconfig.json#L22-L25) to import from `@/routes/xyz` or `@routes/xyz`.

Reply:
I've tested it with this package: https://github.com/npm/validate-npm-package-name and it seems a package like `@data/something` would be valid so there could indeed be overlaps. Wdyt about only using `@/` @fmrsabino @hectorgomezv ?

Reply:
@usame-algan @iamacook do you have any references on what the issue would be if there's a package named `data` and we have a mapping for `@data`?

Reply:
I don't think its a huge issue because it would require us to install a package with such an overlap. Nevertheless I went ahead an installed a package called `@routes/health` just to see how my IDE would behave. There is no error for existing `@routes` imports. When I try to import something from `@routes/...` it will actually suggest both the external package (first suggestion) and the internal paths (everything else).
<img width="752" alt="Screenshot 2023-08-29 at 16 00 29" src="https://github.com/safe-global/safe-client-gateway-nest/assets/5880855/c841af40-3557-4dd5-af25-5804fff1943b">

Considering this imo we can also keep it as `@domain` etc.

Reply:
I'd be interested in knowing how it resolves those conflicts.

While I see the value of just having an `@` I think we get a lot more "semantic" meaning by having the `@domain`, `@data` or `@routes`.

What do you think on, given the above, removing the `@` rule. Meaning that anything that is not under domain, data or routes we should consider adding another "module" (no need to tackle it in this PR though).

Reply:
> While I see the value of just having an @ I think we get a lot more "semantic" meaning by having the @domain, @data or @routes.

I think the inclusion of `/` gives a clear distinction between packages (as they cannot have `/` in their name) vs. aliases but I understand that it is personal preference.

> What do you think on, given the above, removing the @ rule. Meaning that anything that is not under domain, data or routes we should consider adding another "module" (no need to tackle it in this PR though).

I see no issue in having a "global" alias which will encompass this and we'd then not need make any future changes.